src/data/input_output.py

```diff
@@ -72,19 +72,43 @@ def route_parameters(self) -> dict[str, Any]:
 
 
 @dataclass
-class PlayerMatch:
+class PlayerComparator(ABC):
+    field_ids: list[str]
     player: Player
     match_player: Player | None = None
-    _field_ids: list[str] | None = None
 
     @property
-    def field_ids(self) -> list[str]:
-        return self._field_ids or []
-
-    @cached_property
+    @abstractmethod
     def diff_field_ids(self) -> list[str] | None:
         """Returns the list of fields amongst the selected fields on which
         the 2 players have a diff. If match is unset, returns None."""
+        ...
+
+    @abstractmethod
+    def update_player_from_match(self, field_ids: list[str]):
+        """Updates the selected fields of the player from the match_player."""
+        ...
+
+
+@dataclass
+class FidePlayerComparator(PlayerComparator):
+    def match_date_differs(self) -> bool:
+        src_date = self.player.date_of_birth
+        match_date = self.match_player.date_of_birth if self.match_player else None
+        if src_date is None:
+            return match_date is not None
+        if match_date is None:
+            return False
+        if src_date.year != match_date.year:
+            return True
+        if (match_date.month, match_date.day) == (1, 1):
+            return False
+        if (src_date.month, src_date.day) == (match_date.month, match_date.day):
+            return False
+        return True
```

Comment:
Could be siplified into:
```suggestion
        return (src_date.month, src_date.day) != (match_date.month, match_date.day)
```